slash/react/src/Form/Radio/Radio.tsx

```diff
@@ -1,49 +1,78 @@
-import { ComponentPropsWithRef, forwardRef } from "react";
+import {
+  ComponentPropsWithoutRef,
+  ComponentPropsWithRef,
+  forwardRef,
+} from "react";
 import type { Option } from "../core";
 import { RadioItem } from "./RadioItem";
+import { RadioCardGroup } from "./RadioCardGroup";
 
 export enum RadioModes {
   classic = "classic",
   default = "default",
   inline = "inline",
+  card = "card",
 }
 
 type Props = Omit<
   ComponentPropsWithRef<typeof RadioItem>,
   "id" | "label" | "className"
 > & {
   options: Option[];
-  mode?: keyof typeof RadioModes;
-};
+} & (
+    | {
+        mode?: "classic" | "default" | "inline";
+      }
+    | {
+        mode: "card";
+        className?: string;
+        orientation?: ComponentPropsWithoutRef<
+          typeof RadioCardGroup
+        >["orientation"];
+      }
+  );
 
 const getClassNameMode = (mode: Props["mode"]) => {
   switch (mode) {
     case RadioModes.classic:
       return "af-form__radio";
     case RadioModes.inline:
       return "af-form__radio-inline";
+    case RadioModes.card:
+      return "af-form__radio-card";
     default:
       return "af-form__radio-custom";
   }
 };
 
 const Radio = forwardRef<HTMLInputElement, Props>(
   (
-    {
-      classModifier,
-      options,
-      value = "",
-      mode = RadioModes.default,
-      children,
-      disabled,
-      ...otherProps
-    },
+    { classModifier, options, value = "", children, disabled, ...otherProps },
     inputRef,
   ) => {
-    const classNameMode = getClassNameMode(mode);
+    const classNameMode = getClassNameMode(otherProps.mode ?? "default");
+
+    // ICI - le if qui permet de rendre un autre composant
+    if (otherProps.mode === "card") {
+      return (
+        <RadioCardGroup
+          className={otherProps.className}
+          options={options}
+          error={classModifier === "error"}
+          disabled={disabled}
+          orientation={otherProps.orientation}
+        >
+          {children}
+        </RadioCardGroup>
+      );
+    }
+
+    const onlyNecessaryProps = { ...otherProps };
+    delete onlyNecessaryProps.mode;
+
     return options.map((option: Option) => (
       <RadioItem
-        {...otherProps}
+        {...onlyNecessaryProps}
```

Comment:
```suggestion
    { classModifier, options, value = "", children, disabled, mode = "default", ...otherProps },
inputRef,
) => {
    const classNameMode = getClassNameMode(mode);

    // ICI - le if qui permet de rendre un autre composant
    if (mode === "card") {
      return (
        <RadioCardGroup
          className={otherProps.className}
          options={options}
          error={classModifier === "error"}
          disabled={disabled}
          orientation={otherProps.orientation}
        >
          {children}
        </RadioCardGroup>
      );
    }

return options.map((option: Option) => (
<RadioItem
        {...otherProps }
```
il me semble que c'est plus simple comme ça, a vérifier.

Reply:
Je ne peux pas faire ça car sinon mon discriminant de type ne fonctionne pas. Et je ne peux différencié le mode card avec la prop orientation

Reply:
C'est étrange ca pas contre.